compliance_tool/cli.py

```diff
@@ -1,57 +1,67 @@
-                                        help="list available rules for a given profile")
-    parser.add_argument("--profile")
+    list_subparsers.add_parser("rules", parents=[profile_parser],
+                               help="list available rules for a given profile")
 
 
-def prepare_parser_scan(subparsers) -> None:
-    parser = subparsers.add_parser("scan",
-                                   help="perform a compliance scan")
+def prepare_parser_scan(cli_sub_parsers, global_parser, profile_parser) -> None:
+    parser = cli_sub_parsers.add_parser("scan", parents=[global_parser, profile_parser],
+                                        help="perform a compliance scan")
     parser.set_defaults(func=scan.execute)
-    parser.add_argument("--profile")
-    parser.add_argument("--control")
-    parser.add_argument("--rule")
 
-    parser.add_argument("--html")
-    parser.add_argument("--json")
+    parser.add_argument("--control", metavar="CONTROL_ID",
+                        help="control identifier")
+    parser.add_argument("--rule", metavar="RULE_ID",
+                        help="rule identifier")
 
+    parser.add_argument("--html", metavar="DESTINATION_FILE",
+                        help="create an HTML report")
+    parser.add_argument("--json", metavar="DESTINATION_FILE",
+                        help="create a JSON report")
 
-def prepare_parser_remediate(subparsers) -> None:
-    parser = subparsers.add_parser("remediate",
-                                   help="perform a system remediation")
+
+def prepare_parser_remediate(cli_sub_parsers, global_parser, profile_parser) -> None:
+    parser = cli_sub_parsers.add_parser("remediate", parents=[global_parser, profile_parser],
+                                        help="perform a system remediation")
     parser.set_defaults(func=remediate.execute)
-    parser.add_argument("--profile")
-    parser.add_argument("--control")
-    parser.add_argument("--rule")
+
+    parser.add_argument("--control", metavar="CONTROL_ID",
+                        help="control identifier")
+    parser.add_argument("--rule", metavar="RULE_ID",
+                        help="rule identifier")
 
 
 def prepare_parsers() -> argparse.ArgumentParser:
-    parser = argparse.ArgumentParser()
-    subparsers = parser.add_subparsers(required=True)
-    parser.add_argument("--input",
-                        help="policy definitions (file)")
-    parser.add_argument("--tailoring",
-                        help="policy tailoring customization (file)")
-    prepare_parser_list(subparsers)
-    prepare_parser_scan(subparsers)
-    prepare_parser_remediate(subparsers)
-    return parser
+    global_parser = argparse.ArgumentParser(add_help=False)
+    global_parser.add_argument("--input", metavar="SOURCE_FILE",
+                               help="policy definitions")
+    global_parser.add_argument("--tailoring", metavar="SOURCE_FILE",
+                               help="policy tailoring customization")
```

Comment:
Do you think that we should somehow hint what format this should be in?

Reply:
Added a list of recognized formats.